llvm/lib/CodeGen/SelectionDAG/DAGCombiner.cpp

```diff
@@ -3755,6 +3755,50 @@ SDValue DAGCombiner::foldSubToUSubSat(EVT DstVT, SDNode *N, const SDLoc &DL) {
   return SDValue();
 }
 
+template <class MatchContextClass>
+static SDValue foldSubCtlzNot(SDNode *N, SelectionDAG &DAG) {
```

Comment:
could you add a brief comment explaining what this function does, primarily the pattern?

Reply:
I added a comment. There are two patterns that are matched. And the replacing subDAG is slightly different (one any-extend when the other one does not).

Reply:
Ping